packages/utils/spatial-utils/src/spatial.js

```diff
@@ -316,3 +316,39 @@ export async function initializeRasterLayersAndChannels(
   const autoImageLayerDefs = await Promise.all(autoImageLayerDefPromises);
   return [autoImageLayerDefs, nextImageLoaders, nextImageMetaAndLayers];
 }
+
+/**
+ * Convert an array of coordinateTransformations objects to a 16-element
+ * plain JS array using Matrix4 linear algebra transformation functions.
+ * @param {object[]} coordinateTransformations List of objects matching the
+ * OME-NGFF v0.4 coordinateTransformations spec.
+ * @returns {number[]} Plain JS array of 16 numbers representing the Matrix4.
+ */
+export function coordinateTransformationsToMatrix(coordinateTransformations) {
+  let mat = (new Matrix4()).identity();
+  if (coordinateTransformations) {
+    // Applied sequentially and in order according to the OME-NGFF v0.4 spec.
+    // Reference: https://ngff.openmicroscopy.org/0.4/#trafo-md
+    coordinateTransformations.forEach((transform) => {
+      if (transform.type === 'translation') {
+        const translation = [...transform.translation];
+        // Add in z dimension needed for Matrix4 scale API.
+        if (!translation[2]) {
+          translation[2] = 0;
+        }
+        const nextMat = (new Matrix4()).translate(translation);
+        mat = mat.multiplyLeft(nextMat);
```

Comment:
Ok so this looks right.  `multiplyLeft` causes things to happen last, which means that every time this is applied "pushes" an operation on the stack.  The only thing is that the above view config has translation first which is traditionally not done (imagine translating first, and then rotating - you'll end up circumscribing the origin, instead of rotating the actual image).  That being said, this is correct from what I can tell.

Reply:
This is mostly a note for future self when we inevitably mess something up.

Reply:
Also looks like there is no rotation option anyway, which is no fun.

Reply:
Yes it seems like the NGFF spec does not support the full set of affine transformations but perhaps a future version will